docs/docs/mail.md

```diff
@@ -1,345 +1,186 @@
+using the [Noise protocol](https://noiseprotocol.org/noise.html) and the
+[SHA-256](https://www.simplilearn.com/tutorials/cyber-security-tutorial/sha-256-algorithm) algorithm.
+
+AES-GCM is an *authenticated* encryption mechanism. Along with encrypting data, the key is also used to calculate a 
+unique hash called a *tag*. Any change to the message will also change the *tag*. This hashing mechanism prevents 
+attackers from changing the contents of a message without the enclave noticing it.
+
+Like all symmetric ciphers, for AES-GCM to work, both sides must know the same key. In cases where the client 
+knows only a public key, Conclave Mail uses the Elliptic-curve Diffie-Hellman algorithm. The elliptic curve used in 
+Conclave Mail is [Curve25519](https://en.wikipedia.org/wiki/Curve25519).
+
+The client knows the target's public key either because the enclave puts its public key into the remote attestation, 
+represented by an [`EnclaveInstanceInfo`](api/-conclave%20-core/com.r3.conclave.common/-enclave-instance-info/index.html)
+object, or because the enclave obtained the key from an earlier Mail item. Two public keys are all that both 
+parties need to compute the same AES key without any man-in-the-middle being able to calculate the same value.
+
+The sender may also have a long-term static public key that the target will recognize. Conclave appends an 
+encryption of this public key and repeat the calculation between the enclave's public key and the static key, 
+generating a new AES-GCM key as a result.
+
+The handshake process maintains an authentication hash that mixes in the hash of the prologue and the various 
+intermediate stages. This authentication hash allows the unencrypted headers to be tamper-proof against the host, as 
+the host will not be able to recalculate it.
+
+The initialization vector for each AES encryption is a counter. To avoid revealing any information, Conclave throws 
+an exception if the Mail item exceeds the limit of the counter.
+
+The encrypted body consists of secure data that you need to transfer between a client and an enclave. Only the
+target enclave can decrypt the encrypted body of a Mail item. The encrypted body consists of a set of 64KB packets. 
+Each packet has its authentication tag.
+
+## Features of Conclave Mail
+
+Conclave Mail provides various features that are useful when building secure applications.
 
 ### Encryption
 
-The encryption key used by the enclave is private to that enclave but stable across restarts and
-enclave upgrades. This means messages encrypted and delivered by older clients can still be decrypted. The format
-uses the respected [Noise protocol framework](https://noiseprotocol.org/) (with AES/GCM and SHA-256), which is
-the same cryptographic framework used by WhatsApp, the Linux kernel WireGuard protocol and I2P.
+The encryption key used by an enclave is private to that enclave. The encryption key is stable across system restarts 
```

Comment:
The encrption key used for Mail by the enclave is no longer stable across system restarts. This was changed in 1.2, so this bit needs to change! Instead say that the enclave uses a random session key which changes each time the enclave restarts.

Reply:
Corrected in [another PR](https://github.com/R3Conclave/conclave-core-sdk/pull/58).